File: db/model/book.go
package db

import "gorm.io/gorm"

type Book struct {
	gorm.Model
	ISBN        string `gorm:"unique;not null"`
	Title       string `gorm:"not null"`
	Language    string `gorm:"not null"`
	TotalCopy   int    `gorm:"not null"`
	Description string
	CreatedBy   uint       `gorm:"not null"`
	Librarian   *Librarian `gorm:"foreignKey:CreatedBy" json:"librarian,omitempty"`
	Authors     []Author   `gorm:"many2many:author_books;"`
	BorrowItems []BorrowItem
	ShelfID     uint
	CategoryID  uint
}

File: book-service/__mocks__/book.ts

const mockBookRequest = {
    id: 1,
    isbn: '9780743273564',
    title: 'Mock Book',
    language: 'English',
    totalCopy: 1,
    shelfId: 1,
    categoryId: 1,
    description: 'Mock description',
    createdAt: new Date(),
    updatedAt: new Date(),
    deletedAt: null,
  };

const mockBookData = {
    ...mockBookRequest,
    id: 1,
    createdAt: new Date(),
    updatedAt: new Date(),
    deletedAt: null,
  };

export { mockBookRequest, mockBookData };  
